node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -59,12 +60,12 @@ sealed abstract class ClientActor extends Actor with BitcoinSLogger {
 
   /**
     * The parameters for the network we are connected to
-    * i.e. [[org.bitcoins.core.config.MainNet]] or [[org.bitcoins.core.config.TestNet3]]
-    * @return
     */
-  def network: NetworkParameters = config.network
+  val network: NetworkParameters = config.network
```

Comment:
Why do we make the change from a `def` to a `val`? Would a `lazy val` not be preferred?

Reply:
My reasoning is:
- try val
- if it crashes, try lazy val
- if it still crashes, use def

There's sometimes initialization issues with vals and lazy vals, defs don't suffer from this. But defs are the least performant, and lazy vals are slightly less performant than vals.